react/src/context/PostHogProvider.tsx

```diff
@@ -50,11 +64,56 @@ export function PostHogProvider({ children, client, apiKey, options }: WithOptio
         }
 
         if (apiKey) {
+            // If the client was already initialized, we'll attempt to return the existing instance
+            // This is likely to occur when someone is developing locally under `React.StrictMode`
+            //
+            // There's no built-in way to detect React.StrictMode, so we set a flag to check if the client was already initialized
+            // and return the existing instance if it was.
+            //
+            // This might also be triggered if someone attempts to change their `apiKey` or
+            // set a different set of `options`. This is handled separately below.
```

Comment:
This approach makes sense to me! I do think adding some tests would be prudent (is the same client returning the second time calling with the same args, does set_config get called correctly, etc).

Otherwise LGTM!

Reply:
Added a couple tests to assert behavior - and found a slight bug :)